storage/store.go

```diff
@@ -141,30 +136,48 @@ func NewStore(logger hclog.Logger, cfg Configuration) (*Store, error) {
 }
 
 func connector(cfg Configuration) (*sql.DB, error) {
-	params := fmt.Sprintf("%s?database=%s&username=%s&password=%s",
-		cfg.Address,
-		cfg.Database,
-		cfg.Username,
-		cfg.Password,
-	)
+	var conn *sql.DB
 
 	if cfg.CaFile != "" {
-		caCert, err := ioutil.ReadFile(cfg.CaFile)
+		caCert, err := os.ReadFile(cfg.CaFile)
 		if err != nil {
 			return nil, err
 		}
 		caCertPool := x509.NewCertPool()
 		caCertPool.AppendCertsFromPEM(caCert)
-		err = clickhouse.RegisterTLSConfig(tlsConfigKey, &tls.Config{RootCAs: caCertPool})
-		if err != nil {
-			return nil, err
-		}
-		params += fmt.Sprintf(
-			"&secure=true&tls_config=%s",
-			tlsConfigKey,
-		)
+
+		conn = clickhouse.OpenDB(&clickhouse.Options{
+			Addr: []string{sanitize(cfg.Address)},
+			Auth: clickhouse.Auth{
+				Database: cfg.Database,
+				Username: cfg.Username,
+				Password: cfg.Password,
+			},
+			Compression: &clickhouse.Compression{
+				Method: clickhouse.CompressionLZ4,
+			},
+			TLS: &tls.Config{
+				RootCAs: caCertPool,
+			},
+		})
+	} else {
+		conn = clickhouse.OpenDB(&clickhouse.Options{
```

Comment:
We can initialize `clickhouse.Options` before line 141. If there is a certificate, we can set it in the `if cfg.CaFile != "" {` branch. After that, a connection can be established.

It will help to avoid code duplication and simplify the code.